common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -1269,6 +1269,12 @@ private void setToCacheRecord(@NonNull final CacheRecord.CacheRecordBuilder targ
                     accountRecord
             );
 
+            if(idTokensForAccount == null || idTokensForAccount.size() == 0)
```

Comment:
When or why might this happen? I think this is an error case and shouldn't really be happening

Reply:
I ran into this case while testing, If an application(binary) is using more than one client id (say 2), and have credentials for same user in both the client ids (maybe for different tenants/clouds)
When we return getAccounts we return accounts for all client id, but tokens are filtered based on client id; so in that case some idTokens are filtered out if they do not belong to the clientId that was passed.

Reply:
Seems a little strange... @iambmelt Do you know if this is expected behavior?

Reply:
This is expected behavior -- idtokens are scoped to a client_id, same as access tokens. If (in the event of dual-stacking) multiple idtokens are found corresponding to a user, only those matching the current client_id should be returned.